In [45]:
import scipy.io as sio
import pandas as pd

def getting_topics (Testing_OutPut,chosen_class , classes):
    output = classes.index(chosen_class)
    Specific_Class_topics = Testing_OutPut[output]
    sorted_output = sorted(Specific_Class_topics.items(), key = lambda kv: kv[1], reverse=True)
    dictionary_sorted_output = dict(sorted_output)
    return dictionary_sorted_output, sorted_output



a = sio.mmread('BBC_Data\\bbc.mtx')
Terms = open('BBC_Data\\bbc.terms', "r")
data = Terms.read()
terms_into_list = data.split("\n")
terms_into_list = terms_into_list[:-1]


docs = open('BBC_Data\\bbc.docs', "r")
data = docs.read()
docs_into_list = data.split("\n")

test = a.todense()
df = pd.DataFrame(test)
df['terms'] = terms_into_list
df.columns = docs_into_list

matrix([[1., 0., 0., ..., 0., 0., 0.],
        [5., 0., 4., ..., 0., 0., 1.],
        [2., 1., 0., ..., 0., 0., 0.],
        ...,
        [0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.]])

In [ ]:
britian = list(filter(lambda x: x.lower().startswith("brit"), terms_into_list)) 
japan = list(filter(lambda x: x.lower().startswith("japa"), terms_into_list)) 
argentina = list(filter(lambda x: x.lower().startswith("argent"), terms_into_list)) 
australia = list(filter(lambda x: x.lower().startswith("austral"), terms_into_list)) 
brazil = list(filter(lambda x: x.lower().startswith("braz"), terms_into_list)) 
canada = list(filter(lambda x: x.lower().startswith("canad"), terms_into_list)) 
#france = print( list(filter(lambda x: x.lower().startswith("franc"), terms_into_list)) )
french = list(filter(lambda x: x.lower().startswith("frenc"), terms_into_list)) 
germany = list(filter(lambda x: x.lower().startswith("german"), terms_into_list)) 
india = list(filter(lambda x: x.lower().startswith("indi"), terms_into_list))[:3] 
indonesian = list(filter(lambda x: x.lower().startswith("indon"), terms_into_list)) 
italy = list(filter(lambda x: x.lower().startswith("ital"), terms_into_list)) 
korea = list(filter(lambda x: x.lower().startswith("kore"), terms_into_list)) 
mexico = list(filter(lambda x: x.lower().startswith("mexi"), terms_into_list)) 
Arabia = list(filter(lambda x: x.lower().startswith("saud"), terms_into_list)) 
South_African = list(filter(lambda x: x.lower().startswith("afric"), terms_into_list))[:2] 
turkey = list(filter(lambda x: x.lower().startswith("turk"), terms_into_list)) 
america = list(filter(lambda x: x.lower().startswith("ameri"), terms_into_list)) 
us =list(filter(lambda x: x.lower().startswith("usa"), terms_into_list)) 
EU = list(filter(lambda x: x.lower().startswith("europ"), terms_into_list))[:2] 
french.append('franc')
group_of_terms = [britian, japan, argentina, australia, brazil, canada, french, germany,
    india, indonesian, italy, korea, mexico, Arabia, South_African, america, us, EU]
britian_count = 0
japan_count = 0
argentina_count = 0
australia_count = 0 
brazil_count = 0
canada_count = 0
french_count = 0
germany_count = 0
india_count = 0
indonesian_count = 0
italy_count = 0
korea_count = 0
mexico_count = 0
Arabia_count = 0
South_African_count = 0
america_count = 0
us_count = 0
EU_count = 0

Counters_Country = [britian_count,japan_count,argentina_count,australia_count,brazil_count,canada_count,french_count,
    germany_count,india_count,indonesian_count,italy_count,korea_count,mexico_count,Arabia_count,South_African_count,america_count,us_count,EU_count]
    
terms_to_look_for = []
for values in group_of_terms:
    terms_to_look_for.extend(values)


print (terms_to_look_for)

In [47]:
classes = ['business', 'entertainment', 'politics', 'sport', 'tech']
Testing_OutPut = []
for values in classes:    
    filter_col = [col for col in df if col.startswith(values)]
    new_dataframe = df[filter_col]

    Results = []
    for x in range(len(filter_col)):
        test = new_dataframe.iloc[:,x]
        test = pd.DataFrame(test)
        test['terms'] = terms_into_list
        test.columns = ['Values', 'Terms']
        test_ordered = test.loc[(test[['Values']] != 0).all(axis=1)]

        list_of_top_values = test_ordered['Terms']
        output = ' '.join(list_of_top_values)
        Results.append(output)

    Test = {}
    for y in Results:
        split = y.split()
        for strings in split:
            if strings in Test:
                for key,value in Test.items():
                    if key == strings:
                        Test[key] = value + 1


            else:
                x = 1
                Test[strings] = x
    Testing_OutPut.append((Test))

In [53]:
classes = ['business', 'entertainment', 'politics', 'sport', 'tech']

business_dict, business_list = getting_topics(Testing_OutPut, 'business', classes)
entertainment_dict, entertainment_list = getting_topics(Testing_OutPut, 'entertainment', classes)
politics_dict, politics_list = getting_topics(Testing_OutPut, 'politics', classes)
sport_dict, sport_list = getting_topics(Testing_OutPut, 'sport', classes)
tech_dict, tech_list = getting_topics(Testing_OutPut, 'tech', classes)

#### Here is looked at each list and look at the top results to get the general topic of said results. 

###### Example Code of Process is Below

In [124]:
business_list[:50]

[('year', 363),
 ('compani', 267),
 ('market', 250),
 ('firm', 244),
 ('month', 207),
 ('expect', 199),
 ('analyst', 184),
 ('two', 174),
 ('on', 173),
 ('chief', 173),
 ('2004', 172),
 ('growth', 170),
 ('govern', 169),
 ('share', 167),
 ('time', 162),
 ('busi', 162),
 ('report', 162),
 ('price', 159),
 ('countri', 156),
 ('continu', 154),
 ('sale', 153),
 ('world', 153),
 ('rise', 150),
 ('bank', 149),
 ('economi', 148),
 ('econom', 144),
 ('increas', 142),
 ('group', 138),
 ('product', 137),
 ('000', 136),
 ('trade', 134),
 ('recent', 134),
 ('ad', 133),
 ('back', 132),
 ('financi', 131),
 ('execut', 131),
 ('plan', 128),
 ('state', 127),
 ('sai', 126),
 ('cost', 125),
 ('three', 124),
 ('2005', 124),
 ('first', 124),
 ('2003', 119),
 ('rate', 119),
 ('demand', 119),
 ('includ', 119),
 ('against', 118),
 ('invest', 118),
 ('industri', 118)]

In [91]:
classes = ['business', 'entertainment', 'politics', 'sport', 'tech']
Testing_OutPut = []
Total = 0
for values in classes:    
    filter_col = [col for col in df if col.startswith(values)]
    new_dataframe = df[filter_col]

    Results = []
    for x in range(len(filter_col)):
        test = new_dataframe.iloc[:,x]
        test = pd.DataFrame(test)
        test['terms'] = terms_into_list
        test.columns = ['Values', 'Terms']
        test_ordered = test.loc[(test[['Values']] != 0).all(axis=1)]

        list_of_top_values = test_ordered['Terms']
        output = ' '.join(list_of_top_values)
        Results.append(output)

    Counter = 0 
    for y in Results:
        split = y.split()
        for strings in split:
            if strings in terms_to_look_for:
                Counter += 1
                break
            else: 
                continue
    Total += Counter

    for y in Results:
        split = y.split()
        for strings in split:
            # for countries in group_of_terms:
            testing_value = 0
            while testing_value >= (len(Counters_Country) - 1):
                if strings in group_of_terms[testing_value]:
                    Counters_Country[testing_value] += 1
                    break
                else:
                    continue


    for y in Results:
        split = y.split()
        if any(z in max(split, terms_to_look_for, key=len) for z in min(terms_to_look_for, split, key=len)) == True:
            testing_value = 0
            while testing_value <= (len(Counters_Country) - 1):
                if any(t in max(split, group_of_terms[testing_value], key=len) for t in min(group_of_terms[testing_value], split, key=len)) == True:
                    Counters_Country[testing_value] += 1
                    break
                else: 
                    testing_value +=1 
                        
    print(values, 'number of references to countries' , Counter)
    print(Total)

business number of references to countries 356
356
entertainment number of references to countries 237
593
politics number of references to countries 262
855
sport number of references to countries 338
1193
tech number of references to countries 232
1425


In [97]:
Countries_Counters_Print_Out = ['britian count','japan count','argentina_count',
    'australia_count','brazil_count', 'canada_count','french_count','germany_count','india_count', 
    'indonesian_count', 'italy_count','korea_count','mexico_count','Arabia_count','South_African_count','america_count',
    'us_count','EU_count',]

totals_test = 0 
print_out_counter = 0 
for values in Counters_Country:
    totals_test += values 
    print(Countries_Counters_Print_Out[print_out_counter], values)
    print_out_counter+=1
print('total counter', totals_test)

britian count 1582
japan count 411
argentina_count 93
australia_count 302
brazil_count 87
canada_count 81
french_count 494
germany_count 287
india_count 408
indonesian_count 10
italy_count 136
korea_count 45
mexico_count 24
Arabia_count 30
South_African_count 121
america_count 427
us_count 29
EU_count 420
total counter 4987


In [122]:
bbc_table_country = pd.DataFrame([Countries_Counters_Print_Out, Counters_Country]).T
bbc_table_country.columns = ['Countries Counter', 'Number of Articles']
#print(bbc_table_country.to_markdown())


In [113]:
a = sio.mmread('BBC_Sports_Data\\bbcsport.mtx')
Terms = open('BBC_Sports_Data\\bbcsport.terms', "r")
data = Terms.read()
terms_into_list = data.split("\n")
terms_into_list = terms_into_list[:-1]


docs = open('BBC_Sports_Data\\bbcsport.docs', "r")
data = docs.read()
docs_into_list = data.split("\n")

test = a.todense()
df = pd.DataFrame(test)
df['terms'] = terms_into_list
df.columns = docs_into_list

In [115]:
britian = list(filter(lambda x: x.lower().startswith("brit"), terms_into_list)) 
japan = list(filter(lambda x: x.lower().startswith("japa"), terms_into_list)) 
argentina = list(filter(lambda x: x.lower().startswith("argent"), terms_into_list)) 
australia = list(filter(lambda x: x.lower().startswith("austral"), terms_into_list)) 
brazil = list(filter(lambda x: x.lower().startswith("braz"), terms_into_list)) 
canada = list(filter(lambda x: x.lower().startswith("canad"), terms_into_list)) 
#france = print( list(filter(lambda x: x.lower().startswith("franc"), terms_into_list)) )
french = list(filter(lambda x: x.lower().startswith("frenc"), terms_into_list)) 
germany = list(filter(lambda x: x.lower().startswith("german"), terms_into_list)) 
india = list(filter(lambda x: x.lower().startswith("indi"), terms_into_list))[:3] 
indonesian = list(filter(lambda x: x.lower().startswith("indon"), terms_into_list)) 
italy = list(filter(lambda x: x.lower().startswith("ital"), terms_into_list)) 
korea = list(filter(lambda x: x.lower().startswith("kore"), terms_into_list)) 
mexico = list(filter(lambda x: x.lower().startswith("mexi"), terms_into_list)) 
Arabia = list(filter(lambda x: x.lower().startswith("saud"), terms_into_list)) 
South_African = list(filter(lambda x: x.lower().startswith("afric"), terms_into_list))[:2] 
turkey = list(filter(lambda x: x.lower().startswith("turk"), terms_into_list)) 
america = list(filter(lambda x: x.lower().startswith("ameri"), terms_into_list)) 
us =list(filter(lambda x: x.lower().startswith("usa"), terms_into_list)) 
EU = list(filter(lambda x: x.lower().startswith("europ"), terms_into_list))[:2] 
french.append('franc')
group_of_terms = [britian, japan, argentina, australia, brazil, canada, french, germany,
    india, indonesian, italy, korea, mexico, Arabia, South_African, america, us, EU]
britian_count = 0
japan_count = 0
argentina_count = 0
australia_count = 0 
brazil_count = 0
canada_count = 0
french_count = 0
germany_count = 0
india_count = 0
indonesian_count = 0
italy_count = 0
korea_count = 0
mexico_count = 0
Arabia_count = 0
South_African_count = 0
america_count = 0
us_count = 0
EU_count = 0

Counters_Country = [britian_count,japan_count,argentina_count,australia_count,brazil_count,canada_count,french_count,
    germany_count,india_count,indonesian_count,italy_count,korea_count,mexico_count,Arabia_count,South_African_count,america_count,us_count,EU_count]
    
terms_to_look_for = []
for values in group_of_terms:
    terms_to_look_for.extend(values)

In [116]:
classes = ['athletics','cricket','football','rugby','tennis']
Testing_OutPut = []
for values in classes:    
    filter_col = [col for col in df if col.startswith(values)]
    new_dataframe = df[filter_col]

    Results = []
    for x in range(len(filter_col)):
        test = new_dataframe.iloc[:,x]
        test = pd.DataFrame(test)
        test['terms'] = terms_into_list
        test.columns = ['Values', 'Terms']
        test_ordered = test.loc[(test[['Values']] != 0).all(axis=1)]

        list_of_top_values = test_ordered['Terms']
        output = ' '.join(list_of_top_values)
        Results.append(output)

    Test = {}
    for y in Results:
        split = y.split()
        for strings in split:
            if strings in Test:
                for key,value in Test.items():
                    if key == strings:
                        Test[key] = value + 1


            else:
                x = 1
                Test[strings] = x
    Testing_OutPut.append((Test))

In [117]:
classes = ['athletics','cricket','football','rugby','tennis']

athletics_dict, athletics_list = getting_topics(Testing_OutPut, 'athletics', classes)
cricket_dict, cricket_list = getting_topics(Testing_OutPut, 'cricket', classes)
football_dict, football_list = getting_topics(Testing_OutPut, 'football', classes)
rugby_dict, rugby_list = getting_topics(Testing_OutPut, 'rugby', classes)
tennis_dict, tennis_list = getting_topics(Testing_OutPut, 'tennis', classes)

#### Here is looked at each list and look at the top results to get the general topic of said results. 

###### Example Code of Process is Below

In [123]:
athletics_list[:50]

[('olymp', 72),
 ('year', 70),
 ('world', 64),
 ('athlet', 63),
 ('champion', 55),
 ('win', 53),
 ('race', 50),
 ('time', 47),
 ('championship', 46),
 ('athen', 42),
 ('european', 41),
 ('indoor', 41),
 ('set', 41),
 ('take', 40),
 ('final', 40),
 ('befor', 40),
 ('sport', 39),
 ('british', 38),
 ('month', 38),
 ('gold', 38),
 ('compet', 38),
 ('run', 37),
 ('great', 37),
 ('go', 35),
 ('on', 35),
 ('record', 34),
 ('britain', 34),
 ('second', 33),
 ('finish', 33),
 ('best', 33),
 ('start', 33),
 ('medal', 32),
 ('mark', 32),
 ('intern', 32),
 ('back', 32),
 ('first', 31),
 ('titl', 31),
 ('train', 31),
 ('get', 31),
 ('trial', 31),
 ('sprinter', 31),
 ('game', 30),
 ('season', 29),
 ('three', 29),
 ('week', 29),
 ('two', 29),
 ('ban', 29),
 ('test', 29),
 ('event', 28),
 ('drug', 28)]

In [118]:
classes = ['athletics','cricket','football','rugby','tennis']
Testing_OutPut = []
Total = 0
for values in classes:    
    filter_col = [col for col in df if col.startswith(values)]
    new_dataframe = df[filter_col]

    Results = []
    for x in range(len(filter_col)):
        test = new_dataframe.iloc[:,x]
        test = pd.DataFrame(test)
        test['terms'] = terms_into_list
        test.columns = ['Values', 'Terms']
        test_ordered = test.loc[(test[['Values']] != 0).all(axis=1)]

        list_of_top_values = test_ordered['Terms']
        output = ' '.join(list_of_top_values)
        Results.append(output)

    Counter = 0 
    for y in Results:
        split = y.split()
        for strings in split:
            if strings in terms_to_look_for:
                Counter += 1
                break
            else: 
                continue
    Total += Counter

    for y in Results:
        split = y.split()
        for strings in split:
            # for countries in group_of_terms:
            testing_value = 0
            while testing_value >= (len(Counters_Country) - 1):
                if strings in group_of_terms[testing_value]:
                    Counters_Country[testing_value] += 1
                    break
                else:
                    continue


    for y in Results:
        split = y.split()
        if any(z in max(split, terms_to_look_for, key=len) for z in min(terms_to_look_for, split, key=len)) == True:
            testing_value = 0
            while testing_value <= (len(Counters_Country) - 1):
                if any(t in max(split, group_of_terms[testing_value], key=len) for t in min(group_of_terms[testing_value], split, key=len)) == True:
                    Counters_Country[testing_value] += 1
                    break
                else: 
                    testing_value +=1 
                        
    print(values, 'number of references to countries' , Counter)
    print(Total)

athletics number of references to countries 78
78
cricket number of references to countries 101
179
football number of references to countries 111
290
rugby number of references to countries 116
406
tennis number of references to countries 89
495


In [119]:
Countries_Counters_Print_Out = ['britian count','japan count','argentina_count',
    'australia_count','brazil_count', 'canada_count','french_count','germany_count','india_count', 
    'indonesian_count', 'italy_count','korea_count','mexico_count','Arabia_count','South_African_count','america_count',
    'us_count','EU_count',]

totals_test = 0 
print_out_counter = 0 
for values in Counters_Country:
    totals_test += values 
    print(Countries_Counters_Print_Out[print_out_counter], values)
    print_out_counter+=1
print('total counter', totals_test)

britian count 92
japan count 6
argentina_count 18
australia_count 123
brazil_count 18
canada_count 4
french_count 84
germany_count 17
india_count 13
indonesian_count 0
italy_count 24
korea_count 0
mexico_count 2
Arabia_count 0
South_African_count 49
america_count 13
us_count 1
EU_count 31
total counter 495


In [121]:
bbc_table_sports_country = pd.DataFrame([Countries_Counters_Print_Out, Counters_Country]).T
bbc_table_sports_country.columns = ['Countries Counter', 'Number of Articles']
print(bbc_table_sports_country.to_markdown())

|    | Countries Counter   |   Number of Articles |
|---:|:--------------------|---------------------:|
|  0 | britian count       |                   92 |
|  1 | japan count         |                    6 |
|  2 | argentina_count     |                   18 |
|  3 | australia_count     |                  123 |
|  4 | brazil_count        |                   18 |
|  5 | canada_count        |                    4 |
|  6 | french_count        |                   84 |
|  7 | germany_count       |                   17 |
|  8 | india_count         |                   13 |
|  9 | indonesian_count    |                    0 |
| 10 | italy_count         |                   24 |
| 11 | korea_count         |                    0 |
| 12 | mexico_count        |                    2 |
| 13 | Arabia_count        |                    0 |
| 14 | South_African_count |                   49 |
| 15 | america_count       |                   13 |
| 16 | us_count            |                    1 |
| 17 | EU_co

#### Topic #1 : Using the BBC dataset 

##### Question 1: For each of the five categories: what are the most common topics among articles?

###### For Businesses the most common topic was current years financial reports. looking at growth reports in specific sectors as well as looking at the return on investments. 

###### For Entertainment the Most common topics are looking at award shows and who place where in specific music or genres. 

###### For politics, the main topic is about the governing body specifically looking at the different factions within the political sphere. They also seem to use the BBC as a reference for their debates and topics on politics

###### For sports, the main topics are wins vs loses, who has won where and why. As well as identifying areas in which a win could be affected such as injuries or prospects of a new player or replacement. 

###### For tech, it seems that the major focus of the tech is the interaction between "People" and the relationship with the technologies that were produced. As well as looking at the different groups of tech firms. 

##### Question 2: Across all categories: how many articles talk about each of the G20 countries?

###### Across all countries of the G20 there were 1425 which referenced or were referencing the G20, of those countries the specific category spread was, business number of references to countries 356, entertainment number of references to countries 237, politics number of references to countries 262, sport number of references to countries 338, tech number of references to countries 232. As shown on the table below is the specific count of each G20 country coming up in the article. The reason for a higher number than what article is because multiple countries are referenced in a single article. 

| Countries Counter   |   Number of Articles |
|:--------------------|---------------------:|
| britian        |                 1582 |
| japan          |                  411 |
| argentina      |                   93 |
| australia      |                  302 |
| brazil         |                   87 |
| canada         |                   81 |
| french         |                  494 |
| germany        |                  287 |
| india          |                  408 |
| indonesian     |                   10 |
| italy          |                  136 |
| korea          |                   45 |
| mexico         |                   24 |
| Arabia         |                   30 |
| South_African  |                  121 |
| america        |                  427 |
| us             |                   29 |
| EU             |                  420 |


#### Question 3: Describe the methodologies you used to solve questions 1 and 2

###### The dataset had already be run on using a form of Linear Discriminant Analysis, the next steps would be to actually contextualize the data that was presented to me. The first step was to actually pull out the mtx file and set in in a way that is readable as well as align the corresponding axis to the correct values this being the classification, or the topic (word) designation. After this it was simply being able to subset into the different classes. This was done through splitting the dataframe into sections then re appending the row labels on to this subset. After this was done the next step was to analyze the words, specifically I removed the words on columns which had 0's then summed the count of rows based upon each column of that subset. this gives me the total number of outputs specifically the results of the most used words / topics attributed to the specific category. In tackling the 2nd question, the issue of speed of my process was wrong, it took too long to iteratively go through each word and reference it back to the specific list of G20 specific words. As a result, is used a faster method which is to look at all the words in total against all the words in G20 list, and then filter out the ones that didn't have said words. From there you then match based upon country, which is tallied by a counter. Because of this it goes through each G20 country lists word, which allows for instances of looking at multiple G20 countries in a single article. 

#### Question 4: What tools did you use for this analysis?

###### Because the LDP was already ran and the matrix was already made, I didn't have to use much in the way of Sklearn or even pyspark. Instead, I used mainly python functions, pandas and SciPy to access the mtx file. 

#### Question 5: How much time did you devote to the exercise? 

###### In total I spent about 7 hours on this project, most of the time was spent actually understanding the files that I had downloaded. As well as optimize the original way I was planning on doing the project. 

#### Question 6: If you had more time how would you strengthen your answer for question 2?

###### With more time, I think trying to figure out specifics in terms of intent on articles with multiple countries would be good to look at. Because sometimes they may just vaguely reference a country, and not really be the main focus of said article. 



### Topic 2: BBC Sports: 

#### Question 1: For each of the five categories: what are the most common topics among articles?

###### For athletics the articles primarily focused on the Olympics as well as placing in the Olympics. On the Olympics specifically looking at racing, indoor sports, and track and field events

###### For Cricket, the topics are focusing on the play of the ball such as the wicket, as well as players and positions I assume being captain and how they're fairing in the game. The overall topic is a basic review and play by play of the game at hand as well as short and small discussions on player achievement / success in the game. 

###### For Football, the topic has to do with specifically world cup action as well as league placements. They also discuss specific teams as well as how they're doing within their league. 

###### For Rugby, topic is more about the state of the game as well as the teams specifically the ones in different countries such as Ireland, France, and England. As well as reactions or write ups on coaches decisions and plays made during the game. 

###### For Tennis the main topic is the Opens, and the seeding of placements. The topics are who's playing who and the past and future champions. They talk about the games as well as the order in which it looks to be happening. As well as player success and fitness how they're doing and their injury time. 

#### Question 2: Across all categories: how many articles talk about each of the G20 countries?

###### There seems to have been only 503 out of 737 articles, these are using the same abbreviations and listings as the previous G20 countries. For Category Breakdown based upon the sports is as follows: athletics number of references to countries 78,cricket number of references to countries 101, football number of references to countries 111, rugby number of references to countries 116, tennis number of references to countries 89. But for the specifics in terms of country articles this is the breakdown: 
| Countries Counter   |   Number of Articles |
|:--------------------|---------------------:|
| britian        |                   92 |
| japan          |                    6 |
| argentina      |                   18 |
| australia      |                  123 |
| brazil        |                   18 |
| canada        |                    4 |
| french        |                   84 |
| germany        |                   17 |
| india          |                   13 |
| indonesian     |                    0 |
| italy          |                   24 |
| korea          |                    0 |
| mexico         |                    2 |
| Arabia         |                    0 |
| South_African  |                   49 |
| america        |                   13 |
| us             |                    1 |
| EU             |                   31 |

#### Question 3: Describe the methodologies you used to solve questions 1 and 2.

###### The dataset had already be run on using a form of Linear Discriminant Analysis, the next steps would be to actually contextualize the data that was presented to me. The first step was to actually pull out the mtx file and set in in a way that is readable as well as align the corresponding axis to the correct values this being the classification, or the topic (word) designation. After this it was simply being able to subset into the different classes. This was done through splitting the dataframe into sections then re appending the row labels on to this subset. After this was done the next step was to analyze the words, specifically I removed the words on columns which had 0's then summed the count of rows based upon each column of that subset. this gives me the total number of outputs specifically the results of the most used words / topics attributed to the specific category. In tackling the 2nd question, the issue of speed of my process was wrong, it took too long to iteratively go through each word and reference it back to the specific list of G20 specific words. As a result, is used a faster method which is to look at all the words in total against all the words in G20 list, and then filter out the ones that didn't have said words. From there you then match based upon country, which is tallied by a counter. Because of this it goes through each G20 country lists word, which allows for instances of looking at multiple G20 countries in a single article. 

#### Question 4: What tools did you use for this analysis?

###### Because the LDP was already ran and the matrix was already made, I didn't have to use much in the way of Sklearn or even pyspark. Instead, I used mainly python functions, pandas and SciPy to access the mtx file. 

#### Question 5: How much time did you devote to the exercise? 

###### This project took only an hour because I followed the same coding process as I did for the BBC_Data.

#### Question 6: If you had more time how would you strengthen your answer for question 2?

###### I believe that this project would be beneficial to look at the relationships between countries competing, specifically try to nail down the articles that are about matches. This is because it could prevent future issues with double counters for countries because it references both teams. Another would be to work on the EU words, since Europo is a term used to ID articles, this could be prevent mislabeling with Europo Cup or some version of a Cup title. 


